### ISSUE TITLE
Can't expose a public tuple type with a singleton via a public function

### ISSUE THREAD

**Issue by manuranga** [ballerina-platform/ballerina-lang#37145]
```ballerina
public type X ["x"];
public function func() returns X {
    return ["x"];
}
```
above gives following compile time error
`ERROR [main.bal:(4:32,4:33)] attempt to expose non-public symbol ''`

**Comment by MaryamZi** [comment]
This was fixed for update 2 in https://github.com/ballerina-platform/ballerina-lang/pull/36959.

**Comment by SasinduDilshara** [comment]
> This was fixed for update 2 in #36959.

Added tests in https://github.com/ballerina-platform/ballerina-lang/pull/37235
